### DISCUSSION TITLE
Error 24 app.Comparewithnumbers is not a functin

### DISCUSSION THREAD

**Discussion by Roulyoung** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#12] [General]
Dear Joonas,

Very excited to have found your script and cant wait to use it! Unfortunately, I got an error message:

Error 24: app.Comparewithnumbers is not a function

Line 231: return app.Comparewithnumbers(a.name, b.name)

I am using CS6 and I downloaded the standalone package you provided and tested everything as you specified. 

I tried to play around a bit with the code on that specific line but since Im not a coder it was quite futile.

Would be very much appreciated if you know what might be the cause of error!

![Schermopname (2)](https://user-images.githubusercontent.com/29717566/147407070-31f4064e-d62f-48ef-9be3-330fac624ef9.png)

**Comment by joonaspaakko** [comment]
@Roulyoung, if I remember correctly this is needed to sort the files alphanumerically and it's a fairly new method in Photoshop. I guess it was added in CC. The quick and dirty fix would be to replace the lines [`230-232`](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/blob/0d32afe7cb985b89ad3801571625977d6546acb6/script/Batch%20Mockup%20Smart%20Object%20Replacement.jsx#L230-L232) with:

```js
return inputFiles;
```

This will ignore any kinda sorting with the input files, so if your mockup has multiple smart objects to replace, it won't be able to make sure the images are processed in the right order, but it should work otherwise.

> I do vaguely remember that in Windows (or Mac) it may get files in alphanumeric order either way.... (or it might've even been certain PS versions).

**This would likely be the best fix**, assuming it works (not tested). It should retain the same behavior provided by `comparewithnumbers()` if you replace the lines [`230-232`](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/blob/0d32afe7cb985b89ad3801571625977d6546acb6/script/Batch%20Mockup%20Smart%20Object%20Replacement.jsx#L230-L232) with it:

> When I have some time I will likely patch the code with this or something like it.

```js

function sortAlphaNum(a,b) {

    var reA = /[^a-zA-Z]/g;
    var reN = /[^0-9]/g;
    var AInt = parseInt(a, 10);
    var BInt = parseInt(b, 10);

    if ( isNaN(AInt) && isNaN(BInt) ) {
        var aA = a.replace(reA, "");
        var bA = b.replace(reA, "");
        if ( aA === bA ) {
            var aN = parseInt(a.replace(reN, ""), 10);
            var bN = parseInt(b.replace(reN, ""), 10);
            return aN === bN ? 0 : aN > bN ? 1 : -1;
        } 
        else {
            return aA > bA ? 1 : -1;
        }
    } 
    else if ( isNaN(AInt) ) {
        return 1;
    } 
    else if ( isNaN(BInt) ) {
        return -1;
    }
    else {
        return AInt > BInt ? 1 : -1;
    }
}

return inputFiles.sort(function (a, b) {
  return sortAlphaNum(a.name, b.name);
});
```

**Reply by Roulyoung** [reply]
It works! 

Thank you so much!

I did not try the sorting code snippet yet. 

Also, I'm not getting the right dimensions of the photo to work. It now seems either take the width or the lenght but does not automatically fully cover the full Smart Object layer. I did not spend any time trying to look into it yet though.

Let you know when I have played around a bit more!

**Reply by joonaspaakko** [reply]
I wonder if this is because of cs6 or just some resolution related thing. If you can send me the files I can take a look at it. I would imagine I'd only need one input file. And you could color fill any unimportant parts of each file for license or privacy reasons if any. My email is in my github profile if needed for the files.

Did you try the standalone example as is? That might answer some questions. If that example works, it's likely some resolution issue in your other input and/or the mockup or something. But if that doesn't work, I would imagine it has something to do with the PS version.

**Comment by Roulyoung** [comment]
Thank you so much! Can not check it atm because I do not have my Photoshop laptop with me but will let you know when I have in the coming days.

Hope you had a wonderful Christmas.

________________________________

**Comment by joonaspaakko** [comment]
Last night I (finally) fixed both of the issues raised in this discussion in v.1.6. (https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/commit/7b1d377b7bedeba892b7b4fd636f1fa6160415c0).

- Fixed: app.comparewithnumbers is not a function (CS6)
- Fixed: Input files not resizing correctly (CS6)
